### PR TITLE
Add `omdb db validate` subcommands

### PR DESCRIPTION
**Pull request by jmpesp**
In order to diagnose if customer-support#57 is caused by _not_ having the fixes for omicron#3866, add a few commands to omdb to validate the contents of the database:

    $ omdb db validate --help
    Validate the contents of the database

    Usage: omdb db validate <COMMAND>

    Commands:
      validate-volume-references     Validate each `volume_references` column in the region snapshots table
      find-deleted-region-snapshots  Find region snapshots that the corresponding Crucible agent says were deleted
      help                           Print this message or the help of the given subcommand(s)

    Options:
      -h, --help  Print help

This commit also adds an environment variable OMDB_FETCH_LIMIT, which overrides the default fetch limit.

### CODE REVIEW

File: dev-tools/omdb/src/bin/omdb/db.rs [hunk 1]
```diff
@@ -1352,6 +1395,288 @@ async fn cmd_db_eips(
...
+                                    )
+                                    .await?
+                            };
+
+                            if snapshot.time_deleted().is_some() {
+                                // This is ok - Nexus currently soft-deletes its
+                                // resource records.
+
+                                eprintln!(
+                                    "region snapshot {} {} {} at {} was deleted, please remove its record",
```
Comment (leftwo) [inline dev-tools/omdb/src/bin/omdb/db.rs:1619]: This one could use a header for DATASET REGION SNAPSHOT or some such to describe these three.

Seeing as the pattern in repeated below, but just the error changes, you could probably always print the triplet, then just customize the message depending on what the error was.

**Review by leftwo — APPROVED** [review]
I like it (and we need it) Just some small suggestions

File: dev-tools/omdb/src/bin/omdb/db.rs [hunk 2]
```diff
@@ -1352,6 +1395,288 @@ async fn cmd_db_eips(
...
+
+                targets
+                    .read_only_targets
+                    .contains(&region_snapshot.snapshot_addr)
+            })
+            .count();
+
+        if matching_volumes != region_snapshot.volume_references as usize {
+            eprintln!(
+                "region snapshot {} {} {} has {} volume references when it should be {}!",
```
[the remark this answers is not in the document]
Comment (jmpesp) [reply dev-tools/omdb/src/bin/omdb/db.rs]: agreed, done in b3f44e479

File: dev-tools/omdb/src/bin/omdb/db.rs [hunk 3]
```diff
@@ -1352,6 +1395,288 @@ async fn cmd_db_eips(
...
+                matching_volumes,
+            );
+        } else {
+            // The volume references are correct, but additionally check to see
+            // deleting is true when matching_volumes is 0. Be careful: in the
+            // snapshot create saga, the region snapshot record is created
+            // before the snapshot's volume is inserted into the DB. There's a
+            // time between these operations that this function would flag that
+            // this region snapshot should have `deleting` set to true.
+
```
[the remark this answers is not in the document]
Comment (jmpesp) [reply dev-tools/omdb/src/bin/omdb/db.rs:1510]: changed that output in b3f44e479

File: dev-tools/omdb/src/bin/omdb/db.rs (hunk 1 above)
Comment (jmpesp) [reply dev-tools/omdb/src/bin/omdb/db.rs]: agreed, changed in 8b1200535

**Pull request merged**